lib/vm.js

```diff
@@ -151,12 +158,54 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+const kLoaderContextInjection = '____loaderContextInjection____';
+function createModuleContext(sandbox = {}) {
+  const context = createContext(sandbox);
+  context[kLoaderContextInjection] = Loader;
+  const loader = runInContext(
+    `new this['${kLoaderContextInjection}']()`, context);
+  delete context[kLoaderContextInjection];
+  contextLoaders.set(context, loader);
+  return createContext(sandbox);
+}
+
+function createModule(src, {
+  url = 'vm:module',
```

Comment:
Should we make these generated URL values unique for debugging?

Reply:
I wasn't sure how to do that based on the input, maybe a counter variable?